internal/controller/sriov_controller.go

```diff
@@ -0,0 +1,287 @@
+			return fmt.Errorf("error while patching %s %s: %w", network.GetObjectKind().GroupVersionKind().String(), client.ObjectKeyFromObject(network), err)
+		}
+	}
+	return nil
+}
+
+// reconcileSRIOVNetworkNodePolicy generates SRIOVNetworkNodePolicy according to defined Rails
+func (r *SRIOVReconciler) reconcileSRIOVNetworkNodePolicy(ctx context.Context, cfg *config.Config) error {
+	logr := log.FromContext(ctx)
+	for _, railDevMaping := range cfg.RailDeviceMapping {
+		network := r.generateSRIOVNetworkNodePolicy(railDevMaping.RailName, railDevMaping.DevName, cfg.SpectrumXNetworks.MTU)
+		logr.Info("Creating OVSNetwork", "name", network.Name)
+		if err := r.Client.Patch(ctx, network, client.Apply, client.ForceOwnership, client.FieldOwner(spectrumXSRIOVControllerName)); err != nil {
+			return fmt.Errorf("error while patching %s %s: %w", network.GetObjectKind().GroupVersionKind().String(), client.ObjectKeyFromObject(network), err)
+		}
+	}
+	return nil
+}
+
+// generateOVSNetwork generates a OVSNetwork object for the given Rail
+func (r *SRIOVReconciler) generateOVSNetwork(rail *config.Rail, mtu uint) *sriovnetworkv1.OVSNetwork {
+	network := &sriovnetworkv1.OVSNetwork{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rail.Name,
+			Namespace: r.SriovObjNamespace,
+			Labels: map[string]string{
+				spectrumXSRIOVControllerName: "",
+			},
+		},
+		Spec: sriovnetworkv1.OVSNetworkSpec{
+			NetworkNamespace: ovsNetworkNamespace,
+			ResourceName:     rail.Name,
+			IPAM:             fmt.Sprintf(`{"type": "nv-ipam", "poolName": "%s", "poolType": "cidrpool"}`, rail.Name),
+			MTU:              mtu,
+			InterfaceType:    ovsNetworkInterfaceType,
+		},
+	}
+
+	network.ObjectMeta.ManagedFields = nil
+	network.SetGroupVersionKind(sriovnetworkv1.GroupVersion.WithKind(ovsNetworkType))
+	return network
+}
+
+// generateSRIOVNetworkNodePolicy generates a SRIOVNetworkNodePolicy object for the given Rail
+func (r *SRIOVReconciler) generateSRIOVNetworkNodePolicy(railName, deviceName string, mtu uint) *sriovnetworkv1.SriovNetworkNodePolicy {
+	nicSelector := &sriovnetworkv1.SriovNetworkNicSelector{
+		PfNames: []string{deviceName},
+	}
+	bridge := &sriovnetworkv1.Bridge{
+		OVS: &sriovnetworkv1.OVSConfig{
+			Bridge: sriovnetworkv1.OVSBridgeConfig{
+				DatapathType: ovsDataPathType,
+			},
+			Uplink: sriovnetworkv1.OVSUplinkConfig{
+				Interface: sriovnetworkv1.OVSInterfaceConfig{
+					Type: ovsNetworkInterfaceType,
+				},
+			},
+		},
+	}
+	nodeState := &sriovnetworkv1.SriovNetworkNodePolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      railName,
+			Namespace: r.SriovObjNamespace,
+			Labels: map[string]string{
+				spectrumXSRIOVControllerName: "",
+			},
+		},
+		// NOTE: Temporary ignore MTU type conversion until we address it in SR-IOV Network Operator
+		//nolint:gosec
+		Spec: sriovnetworkv1.SriovNetworkNodePolicySpec{
+			ResourceName: railName,
+			Mtu:          int(mtu),
+			NumVfs:       1,
```

Comment:
Not sure why its always 1, maybe its best to add a comment why this is the case

Reply:
done